### PR TITLE
Move all listeners to SLCore.Listeners assembly

### LINKED ISSUE #5244: Move all listeners to SLCore.Listeners assembly

### PR DESCRIPTION
**Pull request by ugras-ergun-sonarsource**
Fixes #5244

### CODE REVIEW

File: src/SLCore.UnitTests/Integration/SLCoreIntegrationSmokeTest.cs [hunk 1]
```diff
@@ -55,8 +52,8 @@ public async Task SLCore_InitializeStandaloneAndShutdown()
 
         var slCoreServiceProvider = new SLCoreServiceProvider(new NoOpThreadHandler(), logger);
         slCoreServiceProvider.SetCurrentConnection(slCoreRunner.Rpc);
-        var slCoreListenerSetUp = new SLCoreListenerSetUp(new[] { new LoggerListener(new TestLogger(logToConsole: true)) });
```
Comment (ugras-ergun-sonarsource) [inline src/SLCore.UnitTests/Integration/SLCoreIntegrationSmokeTest.cs:58]: LoggerListener was not accessible so I commented out the line. Test is ignored anyway.

Comment (georgii-borovinskikh-sonarsource) [reply src/SLCore.UnitTests/Integration/SLCoreIntegrationSmokeTest.cs:58]: Why not fix it? You can add internalsvisibleto easily

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

File: src/SLCore.UnitTests/Integration/SLCoreIntegrationSmokeTest.cs (hunk 1 above)
Comment (ugras-ergun-sonarsource) [reply src/SLCore.UnitTests/Integration/SLCoreIntegrationSmokeTest.cs:58]: it's going to be mostly rewritten in next sprint anyway.

Comment (georgii-borovinskikh-sonarsource) [reply src/SLCore.UnitTests/Integration/SLCoreIntegrationSmokeTest.cs:58]: And you will still need that internalsvisibleto for that assembly 😒

**Pull request merged**